python/lsst/ip/diffim/makeKernel.py

```diff
@@ -0,0 +1,336 @@
+                                        targetFwhmPix=targetFwhmPix,
+                                        referenceFwhmPix=referenceFwhmPix,
+                                        basisDegGauss=basisDegGauss,
+                                        basisSigmaGauss=basisSigmaGauss,
+                                        metadata=metadata)
+        if targetFwhmPix == referenceFwhmPix:
+            self.log.info("Target and reference psf fwhms are equal, falling back to config values")
+        elif referenceFwhmPix > targetFwhmPix:
+            self.log.info("Reference psf fwhm is the greater, normal convolution mode")
+        else:
+            self.log.info("Target psf fwhm is the greater, deconvolution mode")
+
+        return basisList
+
+    def _buildCellSet(self, templateMaskedImage, scienceMaskedImage, candidateList):
+        """Build a SpatialCellSet for use with the solve method.
+
+        Parameters
+        ----------
+        templateMaskedImage : `lsst.afw.image.MaskedImage`
+            MaskedImage to PSF-matched to scienceMaskedImage
+        scienceMaskedImage : `lsst.afw.image.MaskedImage`
+            Reference MaskedImage
+        candidateList : `list`
+            A list of footprints/maskedImages for kernel candidates;
+
+            - Currently supported: list of Footprints or measAlg.PsfCandidateF
+
+        Returns
+        -------
+        kernelCellSet : `lsst.afw.math.SpatialCellSet`
+            a SpatialCellSet for use with self._solve
+        """
+        if not candidateList:
+            raise RuntimeError("Candidate list must be populated by makeCandidateList")
+
+        sizeCellX, sizeCellY = self._adaptCellSize(candidateList)
+
+        # Object to store the KernelCandidates for spatial modeling
+        kernelCellSet = lsst.afw.math.SpatialCellSet(templateMaskedImage.getBBox(),
+                                                     sizeCellX, sizeCellY)
+
+        ps = lsst.pex.config.makePropertySet(self.kConfig)
+        # Place candidates within the spatial grid
+        for cand in candidateList:
+            if isinstance(cand, lsst.afw.detection.Footprint):
+                bbox = cand.getBBox()
+            else:
+                bbox = cand['footprint'].getBBox()
+            tmi = lsst.afw.image.MaskedImageF(templateMaskedImage, bbox)
+            smi = lsst.afw.image.MaskedImageF(scienceMaskedImage, bbox)
+
+            if not isinstance(cand, lsst.afw.detection.Footprint):
+                if 'source' in cand:
+                    cand = cand['source']
+            xPos = cand.getCentroid()[0]
+            yPos = cand.getCentroid()[1]
+            cand = diffimLib.makeKernelCandidate(xPos, yPos, tmi, smi, ps)
+
+            self.log.debug("Candidate %d at %f, %f", cand.getId(), cand.getXCenter(), cand.getYCenter())
+            kernelCellSet.insertCandidate(cand)
+
+        return kernelCellSet
+
+    def _adaptCellSize(self, candidateList):
+        """NOT IMPLEMENTED YET.
```

Comment:
Is this true? There's code below.

Reply:
Yes, the code below returns the input cell sizes without updating them, which is what this method would do.